exporter/exporterhelper/batch_sender_test.go

```diff
@@ -439,6 +439,63 @@ func TestBatchSender_WithBatcherOption(t *testing.T) {
 	}
 }
 
+// TestBatchSender_ShutdownDeadlock tests that the exporter does not deadlock when shutting down while a batch is being
+// merged.
+func TestBatchSender_ShutdownDeadlock(t *testing.T) {
+	blockMerge := make(chan struct{})
+	waitMerge := make(chan struct{}, 10)
+
+	// blockedBatchMergeFunc blocks until the blockMerge channel is closed
+	blockedBatchMergeFunc := func(_ context.Context, r1 Request, _ Request) (Request, error) {
+		waitMerge <- struct{}{}
+		<-blockMerge
+		return r1, nil
+	}
+
+	bCfg := exporterbatcher.NewDefaultConfig()
+	bCfg.FlushTimeout = 10 * time.Minute // high timeout to avoid the timeout to trigger
+	be, err := newBaseExporter(defaultSettings, defaultDataType, newNoopObsrepSender,
+		WithBatcher(bCfg, WithRequestBatchFuncs(blockedBatchMergeFunc, fakeBatchMergeSplitFunc)))
+	require.NoError(t, err)
+	require.NoError(t, be.Start(context.Background(), componenttest.NewNopHost()))
+
+	sink := newFakeRequestSink()
+
+	// Send 10 concurrent requests and wait for them to start
+	startWG := sync.WaitGroup{}
+	for i := 0; i < 10; i++ {
+		startWG.Add(1)
+		go func() {
+			startWG.Done()
```

Comment:
```suggestion
			defer startWG.Done()
```
This should be after the `be.send` call, right? Maybe use defer instead? Or are you purposefully wanting to continue in the main goroutine before send is complete?

Reply:
No, it should be before send. We need to ensure all the goroutines are started. The send is blocking

Reply:
> We need to ensure all the goroutines are started. The send is blocking

The goroutines will all be started regardless of where the `startWG.Done()` call is made, if I'm following correctly. The only impact is when the main goroutine will continue after the `startWG.Wait()` call. I'm wondering if it should wait until all sends are complete or not?

Reply:
the send operations cannot complete until they are unblocked, which we do after the shutdown - that's the purpose of the test case

Reply:
Okay, thanks for clarifying 👍